das/redis_storage_service.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package das
+
+import (
+	"context"
+	"crypto/hmac"
+	"fmt"
+	"regexp"
+	"time"
+
+	"golang.org/x/crypto/sha3"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/pkg/errors"
+	flag "github.com/spf13/pflag"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type RedisConfig struct {
+	RedisUrl   string        `koanf:"redis-url"`
+	Expiration time.Duration `koanf:"redis-expiration"`
+	KeyConfig  string        `koanf:"redis-key-config"`
+}
+
+var DefaultRedisConfig = RedisConfig{
+	RedisUrl:   "",
+	Expiration: time.Hour,
+	KeyConfig:  "",
+}
+
+func RedisConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.String(prefix+".redis-url", DefaultRedisConfig.RedisUrl, "Redis url")
+	f.Duration(prefix+".redis-expiration", DefaultRedisConfig.Expiration, "Redis expiration")
+	f.String(prefix+".redis-key-config", DefaultRedisConfig.KeyConfig, "Redis key config")
+}
+
+type RedisStorageService struct {
+	baseStorageService StorageService
+	redisConfig        RedisConfig
+	signingKey         common.Hash
+	client             redis.UniversalClient
+}
+
+var keyIsHexRegex = regexp.MustCompile("^(0x)?[a-fA-F0-9]{64}$")
+
+func NewRedisStorageService(redisConfig RedisConfig, baseStorageService StorageService) (StorageService, error) {
+	redisOptions, err := redis.ParseURL(redisConfig.RedisUrl)
+	if err != nil {
+		return nil, err
+	}
+	keyIsHex := keyIsHexRegex.Match([]byte(redisConfig.KeyConfig))
+	if !keyIsHex {
+		return nil, errors.New("signing key file contents are not 32 bytes of hex")
+	}
+	signingKey := common.HexToHash(redisConfig.KeyConfig)
```

Comment:
Instead of using regex above, a bit more straightforward to simply validate the return value?
```
if signingKey == common.Hash{} {
    return nil, errors.New("signing key file contents are not 32 bytes of hex")
}
```